src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extension.tsx

```diff
@@ -46,6 +51,7 @@ export interface QueryEditorExtensionDependencies {
   fetchStatus?: ResultStatus;
 }
 
+// When updating this please update docs/plugins/data/query-editor-enhancements.md
 export interface QueryEditorExtensionConfig {
```

Comment:
Nit: Is there a reason why all of the `@returns` are the same?

Reply:
they all return a react component. i could've reworded it but i think the `getX` name should already make it clear what type of component it is referencing. I just reworded it because `The component the query editor extension` doesn't make sense :D